suave/sol/libraries/Suave.sol

```diff
@@ -69,105 +64,96 @@ library Suave {
+    function confidentialStoreStore(BidId bidId, string memory key, bytes memory data) internal view {
+        (bool success, bytes memory data) = CONFIDENTIAL_STORE_STORE.staticcall(abi.encode(bidId, key, data));
         if (!success) {
-            return (false, 0);
+            revert PeekerReverted(CONFIDENTIAL_STORE_STORE, data);
         }
-        return (success, abi.decode(simResults, (uint64)));
     }
 
     function extractHint(bytes memory bundleData) internal view returns (bytes memory) {
-		require(isOffchain());
-
+        require(isOffchain());
         (bool success, bytes memory data) = EXTRACT_HINT.staticcall(abi.encode(bundleData));
-
         if (!success) {
             revert PeekerReverted(EXTRACT_HINT, data);
         }
+
         return data;
     }
 
-	// Builds a block based on the single bid passed in.
-	// Returns a bid which contains the built block and can be queried.
-	// The input bid may contain some intermediate cached results
-	// speeding up the subsequent block building calls.
-	function buildEthBlock(BuildBlockArgs memory blockArgs, BidId bid, string memory namespace) internal view returns (bytes memory, bytes memory) {
-        (bool success, bytes memory builderBid) = BUILD_ETH_BLOCK_PEEKER.staticcall(
-            abi.encode(blockArgs, bid, namespace)
-        );
+    function fetchBids(uint64 cond, string memory namespace) internal view returns (Bid[] memory) {
+        (bool success, bytes memory data) = FETCH_BIDS.staticcall(abi.encode(cond, namespace));
         if (!success) {
-            revert PeekerReverted(BUILD_ETH_BLOCK_PEEKER, builderBid);
+            revert PeekerReverted(FETCH_BIDS, data);
         }
-        return abi.decode(builderBid, (bytes, bytes));
-    }
 
-    function submitEthBlockBidToRelay(string memory relayUrl, bytes memory builderBid) internal view returns (bool, bytes memory) {
-		require(isOffchain());
+        return abi.decode(data, (Bid[]));
+    }
 
-        (bool success, bytes memory err) = SUBMIT_ETH_BLOCK_BID_TO_RELAY.staticcall(
-            abi.encode(relayUrl, builderBid)
-        );
+    function newBid(uint64 decryptionCondition, address[] memory allowedPeekers, string memory bidType)
+        internal
+        view
+        returns (Bid memory)
+    {
+        (bool success, bytes memory data) = NEW_BID.staticcall(abi.encode(decryptionCondition, allowedPeekers, bidType));
+        if (!success) {
+            revert PeekerReverted(NEW_BID, data);
+        }
 
-        return (success, err);
+        return abi.decode(data, (Bid));
     }
-}
 
-function idsEqual(Suave.BidId _l, Suave.BidId _r) pure returns (bool) {
-    bytes memory l = abi.encodePacked(_l);
-    bytes memory r = abi.encodePacked(_r);
-    for (uint i = 0; i < l.length; i++) {
-        if (bytes(l)[i] != r[i]) {
-            return false;
+    function simulateBundle(bytes memory bundleData) internal view returns (uint64) {
```

Comment:
In the new paradigm of peekers requiring success, simulateBundle should return (uint64, bool), and the boolean should mark whether simulation succeeded

Reply:
This will be done in a followup PR.